controllers/subject.js

```diff
@@ -1,7 +1,11 @@
 const subjectService = require('~/services/subject')
 
 const getSubjects = async (req, res) => {
-  const subjects = await subjectService.getSubjects()
+  const { skip, limit, name } = req.query
+
+  const searchFilter = name.trim().length ? { name: { $regex: name, $options:'i' } } : {}
```

Comment:
we have func getRegex you can try to use it here

Reply:
i have refactored